### PR TITLE
STM32: Convert dma users to new dma helpers macros

### PR DESCRIPTION
**Pull request by erwango**
New DT_DMA helpers macros are available to directly access dma controller devices.

Make use of these macros in existing users.

### CODE REVIEW

File: drivers/flash/flash_stm32_qspi.c [hunk 1]
```diff
@@ -695,14 +694,9 @@ static int flash_stm32_qspi_init(const struct device *dev)
...
-	if (dev_data->dma.name != NULL) {
-		dev_data->dma.dev = device_get_binding(dev_data->dma.name);
-		if (!dev_data->dma.dev) {
-			LOG_ERR("%s device not found", dev_data->dma.name);
-			return -ENODEV;
-		}
-	} else {
-		return -EINVAL;
+	if (!device_is_ready(dev_data->dma.dev)) {
+		LOG_ERR("%s device not found", dev_data->dma.dev->name);
```
Comment (galak) [inline drivers/flash/flash_stm32_qspi.c:698]: Message should probably be `device is not ready` or something like that.

**Review by galak — COMMENTED** [review]
LGTM, Just the one comment on the flash driver.

File: drivers/serial/uart_stm32.c [hunk 2]
```diff
@@ -1147,20 +1147,14 @@ static int uart_stm32_async_init(const struct device *dev)
...
 	data->uart_dev = dev;
 
-	if (data->dma_rx.dma_name != NULL) {
-		data->dev_dma_rx = device_get_binding(data->dma_rx.dma_name);
-		if (data->dev_dma_rx == NULL) {
-			LOG_ERR("%s device not found", data->dma_rx.dma_name);
-			return -ENODEV;
-		}
+	if (data->dma_rx.dma_dev != NULL) {
+		__ASSERT_NO_MSG(device_is_ready(data->dma_rx.dma_dev));
```
Comment (shlomow) [inline drivers/serial/uart_stm32.c:1151]: The test of uart async api not working. When I made this change:
```
if(!device_is_ready(data->dma_rx.dma_dev)) {
	return -ENODEV;
}
```
It worked

File: drivers/serial/uart_stm32.c [hunk 3]
```diff
@@ -1147,20 +1147,14 @@ static int uart_stm32_async_init(const struct device *dev)
...
 	}
 
-	if (data->dma_tx.dma_name != NULL) {
-		data->dev_dma_tx = device_get_binding(data->dma_tx.dma_name);
-		if (data->dev_dma_tx == NULL) {
-			LOG_ERR("%s device not found", data->dma_tx.dma_name);
-			return -ENODEV;
-		}
+	if (data->dma_tx.dma_dev != NULL) {
+		__ASSERT_NO_MSG(device_is_ready(data->dma_tx.dma_dev));
```
Comment (shlomow) [inline drivers/serial/uart_stm32.c:1156]: same here

**Review by shlomow — CHANGES_REQUESTED** [review]

File: drivers/serial/uart_stm32.c (hunk 2 above)
Comment (erwango) [reply drivers/serial/uart_stm32.c:1151]: Strange, this was tested ok. I'll have a new look.
Thanks for reporting.

File: drivers/flash/flash_stm32_qspi.c (hunk 1 above)
Comment (erwango) [reply drivers/flash/flash_stm32_qspi.c:698]: Right, fixed

File: drivers/serial/uart_stm32.c (hunk 2 above)
Comment (erwango) [reply drivers/serial/uart_stm32.c:1151]: @shlomow can you check status using https://github.com/zephyrproject-rtos/zephyr/pull/32716 ?

Comment (erwango) [reply drivers/serial/uart_stm32.c:1151]: @shlomow, can  you check status using https://github.com/zephyrproject-rtos/zephyr/pull/32716?

**Review by galak — APPROVED** [review]

**Comment by erwango** [comment]
DNM waiting for @shlomow's feedback

**Review by FRASTM — APPROVED** [review]

**Comment by shlomow** [comment]
@erwango It doesn't work. I think that you are returning `-ENODEV` regardless the device is ready.

**Comment by erwango** [comment]
> @erwango It doesn't work. I think that you are returning `-ENODEV` regardless the device is ready.

This is indeed obvious (today...). Thanks!

File: drivers/serial/uart_stm32.c (hunk 2 above)
Comment (erwango) [reply drivers/serial/uart_stm32.c]: Fixed, txs

**Comment by shlomow** [comment]
@erwango work for me now. Thanks!

**Pull request merged**